src/Orleans.Runtime/Streams/QueueBalancer/LeaseBasedQueueBalancer.cs

```diff
@@ -174,41 +171,44 @@ public override IEnumerable<QueueId> GetMyQueues()
 
         private async Task ReleaseLeasesToMeetResponsibility()
         {
-            var queueCountToRelease = this.myQueues.Count - this.maximumRespobsibility;
+            var queueCountToRelease = this.myQueues.Count - this.maximumResponsibility;
             if (queueCountToRelease <= 0)
                 return;
             var queuesToGiveUp = this.myQueues.GetRange(0, queueCountToRelease);
             await this.leaseProvider.Release(LeaseCategory, queuesToGiveUp.Select(queue => queue.AcquiredLease).ToArray());
             //remove queuesToGiveUp from myQueue list after the balancer released the leases on them
             this.myQueues.RemoveRange(0, queueCountToRelease);
-            this.logger.Info($"ReleaseLeasesToMeetResponsibility: released {queueCountToRelease} queues, current queue Count: {this.myQueues.Count}");
+            this.logger.Info($"Released leases for {queueCountToRelease} queues");
+            this.logger.LogInformation($"I now own leases for {this.myQueues.Count} of an expected {this.minimumResponsibility} to {this.maximumResponsibility} queues.");
         }
 
-        private Task AcquireLeaseToMeetMaxResponsibilty(object state)
+        private Task AcquireLeaseToMeetMaxResponsibility(object state)
         {
-            return AcquireLeasesToMeetExpectation(this.maximumRespobsibility);
+            return AcquireLeasesToMeetExpectation(this.maximumResponsibility);
         }
 
         private Task AcquireLeasesToMeetMinResponsibility()
         {
-            return AcquireLeasesToMeetExpectation(this.minimumResponsibilty);
+            return AcquireLeasesToMeetExpectation(this.minimumResponsibility);
         }
 
         private async Task AcquireLeasesToMeetExpectation(int expectedTotalLeaseCount)
         {
             int maxAttempts = 5;
             int attempts = 0;
             int leasesToAquire = expectedTotalLeaseCount - this.myQueues.Count;
-            this.logger.Info($"AcquireLeasesToMeetExpectation : Try to acquire {leasesToAquire} queues");
-            while (attempts ++ <= maxAttempts && leasesToAquire > 0)
+            if (leasesToAquire <= 0) return;
```

Comment:
you don't need this line, the while loop already check this

Reply:
This prevents log spam when nothing is changing.  Otherwise the log line at the end of this function is fired when nothing is done.